### PR TITLE
Add TLS and Path Routing Tests for Nginx, Voyager and Traefik

### PR DESCRIPTION
**Pull request by xiancao**
Add TLS and Path routing tests for Nginx, Voyager and Traefik. Also add the two domains routing test for Nginx. 

Jenkins run:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2098/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2102/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2142/

### CODE REVIEW

**Review by sankarpn — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 1]
```diff
@@ -460,36 +503,76 @@ public void testTraefikHostHttpsRoutingAcrossDomains() {
...
+      verifyClusterLoadbalancing(domainUid, domainUid + "." + defaultNamespace + ".voyager.nonssl.test",
+          "http", getVoyagerLbNodePort(ingressName, "tcp-80"),
+          replicaCount, true, "");
+    }
+  }
+
+  /**
+   * Test verifies multiple WebLogic domains can be loadbalanced by Voyager loadbalancer with tls based routing rules.
+   * Accesses the clusterview application deployed in the WebLogic cluster through Voyager loadbalancer and verifies it
+   * is correctly routed to the specific domain cluster identified by the -H host header.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:524]: Do we need to provide  host header for path routing

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 2]
```diff
@@ -460,36 +503,76 @@ public void testTraefikHostHttpsRoutingAcrossDomains() {
...
    * Configure the Apache webtier as a load balancer for a WebLogic domain using the default configuration.
    * It only support HTTP protocol.
    * For details, please see
    * https://github.com/oracle/weblogic-kubernetes-operator/tree/master/kubernetes/samples/charts/apache-samples/default-sample
    */
-  @Order(8)
+  @Order(11)
   @Test
   @DisplayName("verify Apache load balancer default sample through HTTP channel")
   public void testApacheDefaultSample() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:570]: Modify Test name to testApacheLoadBalancingDefaultSample()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 3]
```diff
@@ -500,7 +583,7 @@ public void testApacheDefaultSample() {
    * For more details, please check:
    * https://github.com/oracle/weblogic-kubernetes-operator/tree/master/kubernetes/samples/charts/apache-samples/custom-sample
    */
-  @Order(9)
+  @Order(12)
   @Test
   @DisplayName("verify Apache load balancer custom sample through HTTP and HTTPS channel")
   public void testApacheCustomSample() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:589]: Modify Test name to testApacheLoadBalancingCutomSample()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 4]
```diff
@@ -509,15 +592,71 @@ public void testApacheCustomSample() {
...
 
+  /**
+   * Test verifies multiple WebLogic domains can be loadbalanced by NGINX loadbalancer with host based routing rules.
+   * Accesses the clusterview application deployed in the WebLogic cluster through NGINX host routing
+   * and verifies it is correctly routed to the specific domain cluster identified by the -H host header.
+   *
+   */
+  @Order(13)
+  @Test
+  @DisplayName("verify NGINX host routing with HTTP across two domains")
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:612]: verify NGINX host routing with HTTP protocol across two domains

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 5]
```diff
@@ -509,15 +592,71 @@ public void testApacheCustomSample() {
...
+  }
+
+  /**
+   * Test verifies multiple WebLogic domains can be loadbalanced by NGINX loadbalancer with path routing rules.
+   * Accesses the clusterview application deployed in the WebLogic cluster through NGINX loadbalancer and verifies it
+   * is correctly routed to the specific domain cluster.
+   */
+  @Order(15)
+  @Test
+  @DisplayName("Load balance WebLogic cluster traffic through NGINX load balancer path routing")
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:649]: Provide consistent description for both host/path routing test methods as follows ...

  @DisplayName("verify NGINX path routing with HTTP protocol across two domains")

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 6]
```diff
@@ -363,6 +377,9 @@ public void testDeployAppAndInstallIngressControllers() {
     voyagerHelmParams =
         installAndVerifyVoyager(voyagerNamespace, cloudProvider, enableValidatingWebhook);
 
+    // install and verify Nginx
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:380]: Modify the description 
  Test deploy applications and install ingress controllers Traefik. NGINX and Voyager.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 7]
```diff
@@ -450,8 +478,23 @@ public void testTraefikHostHttpsRoutingAcrossDomains() {
...
+
+  /**
+   * Test verifies multiple WebLogic domains can be loadbalanced by Traefik loadbalancer with path routing rules.
+   */
+  @Order(7)
+  @Test
+  @DisplayName("Loadbalance WebLogic cluster traffic through Traefik loadbalancer path routing")
+  public void testTraefikPathRoutingAcrossDomains() {
+
+    logger.info("Verifying https traffic");
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:494]: Modify -- > logger.info("Verifying Path Routing using http protocol");

**Review by anpanigr — COMMENTED** [review]
Few more  review  comments 
a. testTwoDomainsManagedByOneOperatorSharingPV
Line # 336   ( Create one operator if it is not running) -  The test is in order. So we should know if the Oprator is running or not
Line # 337 (Create domain domain1 and domain2 dynamic cluster in default namespace, managed by operator1.) should be modified to "Create domain domain1 and domain2 with a dynamic cluster in each domain in default namespace, managed by operator1"
 (Shut down domain1 and back up, plus verify no impact on domain2.)  -->  Shut down domain1 and make sure there is impact on domain2.) 

Line #436 Modify the description 
 Verify WebLogic console is accessible Teaefik LoadBalancer Web Port"

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:524]: Those are for host https routing.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java [hunk 8]
```diff
@@ -166,6 +167,9 @@
       .orElse("11.1.5");
   public static final String GRAFANA_CHART_VERSION = Optional.ofNullable(System.getenv("GRAFANA_CHART_VERSION"))
       .orElse("5.0.20");
+  public static final String STABLE_REPO_NAME = "stable";
```
[the remark this answers is not in the document]
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java [hunk 9]
```diff
@@ -166,6 +167,9 @@
...
+  public static final String GOOGLE_REPO_URL = "https://kubernetes-charts.storage.googleapis.com/";
```
[the remark this answers is not in the document]
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 10]
```diff
@@ -384,21 +401,32 @@ public void testDeployAppAndInstallIngressControllers() {
...
           defaultNamespace, tlsKeyFile, tlsCertFile));
     }
-    // create loadbalancing rules for Traefik
+    // create loadbalancing rules for Traefik, Voyager and NGINX
     createTraefikIngressRoutingRules();
-    createVoyagerIngressRoutingRules();
+
+    // create ingress with non-tls host routing
+    createVoyagerIngressHostRoutingRules(false);
+    createNginxIngressHostRouting4TwoDomains(false);
```
[the remark this answers is not in the document]
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 11]
```diff
@@ -384,21 +401,32 @@ public void testDeployAppAndInstallIngressControllers() {
...
+
+    // create ingress with tls host routing
+    createVoyagerIngressHostRoutingRules(true);
+    createNginxIngressHostRouting4TwoDomains(true);
+
+    // create ingress with path routing
+    createVoyagerIngressPathRoutingRules();
+    createNginxIngressPathRouting4TwoDomains();
```
[the remark this answers is not in the document]
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 5 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 6 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:379]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 7 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java]: fixed

**Review by anpanigr — APPROVED** [review]
LGTM

**Pull request merged**